presto-function-namespace-managers/src/main/java/com/facebook/presto/functionNamespace/json/JsonFileBasedFunctionDefinitionProvider.java

```diff
@@ -15,24 +15,54 @@
 
 import com.facebook.airlift.log.Logger;
 
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
 import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 import static com.facebook.presto.plugin.base.JsonUtils.parseJson;
+import static java.util.Locale.ENGLISH;
 
 public class JsonFileBasedFunctionDefinitionProvider
         implements FunctionDefinitionProvider
 {
     private static final Logger log = Logger.get(JsonFileBasedFunctionDefinitionProvider.class);
+    private static final int MAX_DIRECTORY_DEPTH = 4;
+    private static final String JSON_FILE_EXTENSION = ".json";
 
     @Override
     public UdfFunctionSignatureMap getUdfDefinition(String filePath)
     {
+        Map<String, List<JsonBasedUdfFunctionMetadata>> udfSignatureMap = new HashMap<>();
+
+        // If any of the files is not a valid source, the function will return null;
```

Comment:
i know you didn't introduce this, and you don't have to change it here, but this isn't a very presto-y way of doing things.  We prefer being super explicit if something could be null and return optional instead, and also generally would throw an exception rather than ignore the malformed files).

Reply:
That's a good point; I noticed the `null` usage; but, I didn't want to make unnecessary changes; I can refactor this in a different PR if needed, BTW.